src/api/batchVerify/index.ts

```diff
@@ -0,0 +1,111 @@
+import { handleTransaction } from '../../utils/transactions';
+import { AccountConnection, WalletConnection } from '../connection/types';
+import { EventEmitter } from 'events';
+import { BatchVerifyTransactionInfo } from '../../types';
+import { VerifyOptions } from '../../session/types';
+import { TransactionType, ZkVerifyEvents } from '../../enums';
+import { format } from '../format';
+import { createSubmitProofExtrinsic } from '../extrinsic';
+import { VerifyInput } from '../verify/types';
+import { SubmittableExtrinsic } from '@polkadot/api/types';
+import { FormattedProofData } from '../format/types';
+import { getKeyringAccountIfAvailable } from '../../utils/helpers';
+
+export const batchVerify = async (
+  connection: AccountConnection | WalletConnection,
+  options: VerifyOptions,
+  emitter: EventEmitter,
+  input: VerifyInput[],
```

Comment:
Isn't it a bit confusing the fact that this `VerifyInput` can be either a ProofData or an already formed extrinsic ?

Reply:
Just part of how its always been on zkverifyjs can use either option

Reply:
The extrinsic for instance is much smaller for than the full proof json, useful for queueing etc